### PR TITLE
Updated ABOUT.md

### PR DESCRIPTION
**Pull request by amrtanair**
corrected the mistakes. Anything else I could change?
Thanks a lot for your input

### CODE REVIEW

File: docs/ABOUT.md [hunk 1]
```diff
@@ -1 +1,29 @@
...
+ - It must be robust and secure.
+- It must be architecture-neutral and portable.
+- It must execute with high performance.
+- It must be interpreted, threaded, and dynamic.
+
+The language derives much of its syntax from C and C++, but it has fewer low-level facilities than either of them.
+
+Why Learn JAVA?
+
+- JAVA IS EVERYWHERE. It is one of the most in-demand programming languages, a standard for enterprise software, web-based content, games and mobile apps, as well as the Android operating system.
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: I suggest some minor edits to the first sentence:
> JAVA IS EVERYWHERE. It is one of the most in-demand programming languages: a standard for enterprise software, web-based content, games and mobile apps, and the Android operating system.

File: docs/ABOUT.md [hunk 2]
```diff
@@ -1 +1,29 @@
...
+- It must execute with high performance.
+- It must be interpreted, threaded, and dynamic.
+
+The language derives much of its syntax from C and C++, but it has fewer low-level facilities than either of them.
+
+Why Learn JAVA?
+
+- JAVA IS EVERYWHERE. It is one of the most in-demand programming languages, a standard for enterprise software, web-based content, games and mobile apps, as well as the Android operating system. 
+
+- Java is Platform Independent: This because of its [Write Once, Run Anywhere](https://en.wikipedia.org/wiki/Write_once,_run_anywhere)  policy.This is still one of the reasons for Java being a relatively better programming language. Most of the Java applications are developed in Windows environment and run on UNIX platform.
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: There should be a space after the period following the word 'policy'.  The second sentence is unclear.  Better than what?  For what?

> This is still one of the reasons for Java being a relatively better programming language. 

The final sentence may be true (I have no idea-- do you have a citation?), but I'm not sure how it helps the reader.  Are you trying to support your point about it being cross-platform?

>Most of the Java applications are developed in Windows environment and run on UNIX platform.

File: docs/ABOUT.md [hunk 3]
```diff
@@ -1 +1,29 @@
...
+- Java has a Rich API:
+Java provides [API](https://en.wikipedia.org/wiki/Application_programming_interface)(Application Programming Interface) for I/O, networking, utilities, XML parsing, database connection, and almost everything. Whatever left is covered by open source libraries like Apache Commons, Google Guava and others.
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: The first sentence should probably start "Java provides [API]s", with a s.  Probably "database connection" should be "database connections".  That sentence should also probably end like so:

>database connections, and many other functions.

'Almost everything' is probably too vague and sweeping.   Finally, you might start the last sentence with: "Standard libraries can be supplemented by open source libraries like "

**Comment by matthewmorgan** [comment]
@amrtanair thanks again for doing this.  I think there just a couple more things that should be tightened up.  Please let me know if you need help.

**Comment by matthewmorgan** [comment]
@amrtanair one thing to know-- it shouldn't necessary to create a new pull request for each set of edits.  You should be able to just edit the existing branch `patch-2` and commit it.

Ideally the all commits for your changes would be squashed in to one-- if I can help you do that please let me know.

**Comment by amrtanair** [comment]
I would really like to know how to do it that way. Could you please tell me
how? I tried editing the existing branch but I think I was going about it
the wrong away. It would be great if you could help me.

Regards,
Amrita H Nair

On Mon, Mar 7, 2016 at 10:57 PM, Matthew Morgan <notifications@github.com>
wrote:

> @amrtanair <https://github.com/amrtanair> one thing to know-- it
> shouldn't necessary to create a new pull request for each set of edits. You
> should be able to just edit the existing branch patch-2 and commit it.
>
> Ideally the all commits for your changes would be squashed in to one-- if
> I can help you do that please let me know.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/exercism/xjava/pull/103#issuecomment-193358990>.
>



-- 
Regards,
Amrita H Nair

**Comment by matthewmorgan** [comment]
Closed by #104

**Comment by matthewmorgan** [comment]
Usually you do things in this order:

fork the exercism/xjava repo, then in your local command line:
```
git clone https://github.com/amrtanair/xjava
//then create and check out a new branch to do your work on.  Give it a sensible name
git branch my-awesome-update
git checkout my-awesome-update
//...make your changes, run the tests as described in the Contributing Guide
//now push your local branch with your local changes back up to your fork on GitHub
git push origin my-awesome-update
```
You can now make the pull request.  We'll review it, suggest changes or ask questions, and you can make those changes locally.  When you're done, commit your new changes to your local branch then roll everything in to one commit.

To combine your new changes with the old:
1) reset your git HEAD to the last commit on master:
```
git log
//you should see something like this:
commit c4b10d7d0b7b30b85843e67adcf0dbf120a432ae
Author: amrtanair <amritaanair98@gmail.com>
Date:   Mon Mar 7 12:38:30 2016 +0530

    Updated ABOUT.md
    
    Made some edits to my changes


commit 00ec4082ed62bfb8ba7f56eaced3b4868efc54f8
Author: amrtanair <amritaanair98@gmail.com>
Date:   Mon Mar 7 10:38:30 2016 +0530

    Updated ABOUT.md
    
    corrected the mistakes

commit b03f4f1d7254b8406e30a4dc68ca53598615e369
Merge: 2becaf4 c4b10d7
Author: Katrina Owen <katrina.owen@gmail.com>
Date:   Sun Mar 6 12:53:46 2016 -0700

    Merge pull request #99 from exercism/new-exercise-simple-linked-list
    
    Add new exercise simple linked list
```

The ref BEFORE the first commit you made (EG the one that reflects the repo when you forked it) is the one to which you want to reset, in this case: b03f4f1d7254b8406e30a4dc68ca53598615e369

You will reset, and then make a new commit with a new message that will roll up all your changes since you forked.

```
git reset --soft b03f4f1
git commit -am 'Updated contents of ABOUT.md'
```

2) Now push to the branch on your fork

```
git push -f origin my-awesome-update
```
When you've taken the final step (`push -f`) your branch on your fork will update, and the PR will be changed automatically.  Travis will re-run the tests, and we'll review again.  Once the review-change cycle is done, we merge the PR.

Does that make sense?

**Comment by amrtanair** [comment]
Yes, that did help me a lot. I had been confused for a long time on how to
roll all my commits into one. I will surely implement this the next time I
contribute :)

Regards,
Amrita H Nair

On Tue, Mar 8, 2016 at 12:10 AM, Matthew Morgan <notifications@github.com>
wrote:

> Usually you do things in this order:
>
> fork the exercism/xjava repo, then in your local command line:
>
> git clone https://github.com/amrtanair/xjava
> //then create and check out a new branch to do your work on.  Give it a sensible name
> git branch my-awesome-update
> git checkout my-awesome-update
> //...make your changes, run the tests as described in the Contributing Guide
> //now push your local branch with your local changes back up to your fork on GitHub
> git push origin my-awesome-update
>
> You can now make the pull request. We'll review it, suggest changes or ask
> questions, and you can make those changes locally. When you're done, commit
> your new changes to your local branch then roll everything in to one commit.
>
> To combine your new changes with the old:
> 1) reset your git HEAD to the last commit on master:
>
> git log
> //you should see something like this:
> commit c4b10d7d0b7b30b85843e67adcf0dbf120a432ae
> Author: amrtanair <amritaanair98@gmail.com>
> Date:   Mon Mar 7 12:38:30 2016 +0530
>
>     Updated ABOUT.md
>
>     Made some edits to my changes
>
>
> commit 00ec4082ed62bfb8ba7f56eaced3b4868efc54f8
> Author: amrtanair <amritaanair98@gmail.com>
> Date:   Mon Mar 7 10:38:30 2016 +0530
>
>     Updated ABOUT.md
>
>     corrected the mistakes
>
> commit b03f4f1d7254b8406e30a4dc68ca53598615e369
> Merge: 2becaf4 c4b10d7
> Author: Katrina Owen <katrina.owen@gmail.com>
> Date:   Sun Mar 6 12:53:46 2016 -0700
>
>     Merge pull request #99 from exercism/new-exercise-simple-linked-list
>
>     Add new exercise simple linked list
>
> The ref BEFORE the first commit you made (EG the one that reflects the
> repo when you forked it) is the one to which you want to reset, in this
> case: b03f4f1
> <https://github.com/exercism/xjava/commit/b03f4f1d7254b8406e30a4dc68ca53598615e369>
>
> You will reset, and then make a new commit with a new message that will
> roll up all your changes since you forked.
>
> git reset --soft b03f4f1
> git commit -am 'Updated contents of ABOUT.md'
>
> 2) Now push to the branch on your fork
>
> git push -f origin my-awesome-update
>
> When you've taken the final step (push -f) your branch on your fork will
> update, and the PR will be changed automatically. Travis will re-run the
> tests, and we'll review again. Once the review-change cycle is done, we
> merge the PR.
>
> Does that make sense?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/exercism/xjava/pull/103#issuecomment-193387420>.
>



-- 
Regards,
Amrita H Nair

**Comment by matthewmorgan** [comment]
Awesome.  Feel free to reach out if you get stuck or have any questions.

**Pull request closed without merging**